Server/src/main/java/app/ModelController.java

```diff
@@ -249,4 +265,21 @@ public Instances createInstancesFromAssets(List<Asset> assets) {
         }
         return data;
     }
+
+    /**
+     * evaluates a model on a separate thread using the evaluation class
+     * @param trainedModel
+     * @author talal
+     */
+    public void modelToEvaluate(TrainedModel trainedModel) throws Exception {
+        trainedModel.setAssetTypeID(1);
+        int trainAssets =trainedModel.getModelStrategy().getTrainsAssets();
+        int testAssets=trainedModel.getModelStrategy().getTestAssets();
```

Comment:
its better to keep track of an ArrayList of assets instead of an int to avoid calling assetDaoImpl.getAssetsFromAssetTypeID(trainedModel.getAssetTypeID()) multiple times for no reason

Reply:
it doesn't work, you can't have a list in model strategy, it caused a nonTransient error, I tried to fix it for hours but it didn't work

Reply:
I believe we fixed the transient error for the most part. However, the changing of int to an arraylist of assets will be postpone to a further issue due to time constraints.